src/test/java/integration/JobRoleIntegrationTest.java

```diff
@@ -30,6 +33,196 @@ void getJobRoles_ShouldReturnListOfJobRoles() {
         Assertions.assertFalse(response.isEmpty());
     }
 
+    @Test
+    void postJobRoles_ShouldReturnIdOfJobRole() {
+
+        JobRoleRequest jobRoleRequest = new JobRoleRequest.Builder()
+                .roleName("Software Engineer")
+                .location("Belfast")
+                .capabilityID(1)
+                .bandID(1)
+                .closingDate(new Date(System.currentTimeMillis()))
+                .description("description")
+                .responsibilities( "responsibilities")
+                .jobSpec("jobSpec")
+                .positions(2)
+                .build();
+
+        Client client = APP.client();
+
+        int id = client
+                .target("http://localhost:8080/api/job-roles")
+                .request()
+                .post(Entity.json(jobRoleRequest))
+                .readEntity((Integer.class));
+
+        Assertions.assertTrue(id > 0);
+    }
+
+    @Test
+    void postJobRoles_RoleNameTooLong_ShouldReturn400() {
```

Comment:
no need for all of these 400 integration tests, just 1 would suffice and test the other scenarios via unit tests